udata/core/dataset/api.py

```diff
@@ -146,6 +146,9 @@ def post(self, dataset):
         badge = DatasetBadge(created=datetime.now(),
                              created_by=current_user.id)
         form.populate_obj(badge)
+        for existing_badge in dataset.badges:
+            if existing_badge.kind == badge.kind:
+                return badge
```

Comment:
It should be existing_badge as return value as its the one which is actually stored